File: assisted-inject-test-java/src/test/java/com/example/assisted/GreetingFactory.java
package com.example.assisted;

import com.pkware.micronaut.assisted.Assisted;
import io.micronaut.aop.Introduction;

/**
 * Factory interface that uses @Assisted from the assisted-inject JAR.
 * <p>
 * Note: @Introduction must be added directly because Java annotation processing
 * doesn't process meta-annotations from compiled JARs. This is the exact issue
 * this integration test is designed to catch.
 */
@Introduction
@Assisted
public interface GreetingFactory {
  Greeting create(String name);
}

File: assisted-inject-test-ksp/src/test/kotlin/com/example/assisted/GreetingFactory.kt
package com.example.assisted

import com.pkware.micronaut.assisted.Assisted

/**
 * Factory interface that uses @Assisted from the assisted-inject JAR.
 *
 * Unlike the Java integration test, this does NOT need @Introduction directly.
 * KSP correctly processes meta-annotations from compiled JARs, so the @Introduction
 * on @Assisted is detected and the factory implementation is generated.
 *
 * This proves Kotlin/KSP users get a cleaner experience - just @Assisted is enough.
 */
@Assisted
interface GreetingFactory {
  fun create(name: String): Greeting
}
